cps/src/main/java/org/jenkinsci/plugins/workflow/cps/steps/ParallelStep.java

```diff
@@ -47,11 +49,11 @@ public StepExecution start(StepContext context) throws Exception {
     }
 
     @PersistIn(FLOW_NODE)
-    public static class ParallelLabelAction extends LabelAction {
+    public static class ParallelLabelAction extends LabelAction implements ThreadNameAction {
         private final String branchName;
 
         ParallelLabelAction(String branchName) {
-            super(null);
+            super(branchName);
```

Comment:
Revert this, passing null to the super call is intentional since we override `getDisplayName` and do not want to store two fields.